server/remote_cache/digest/digest.go

```diff
@@ -39,25 +39,35 @@ var (
 	// - "blobs/469db13020c60f8bdf9c89aa4e9a449914db23139b53a24d064f967a51057868/39120"
 	// - "blobs/ac/469db13020c60f8bdf9c89aa4e9a449914db23139b53a24d064f967a51057868/39120"
 	// - "uploads/2042a8f9-eade-4271-ae58-f5f6f5a32555/blobs/8afb02ca7aace3ae5cd8748ac589e2e33022b1a4bfd22d5d234c5887e270fe9c/17997850"
-	uploadRegex      = regexp.MustCompile("^(?:(?:(?P<instance_name>.*)/)?uploads/(?P<uuid>[a-f0-9-]{36})/)?blobs/(?P<hash>[a-f0-9]{64})/(?P<size>\\d+)")
-	downloadRegex    = regexp.MustCompile("^(?:(?P<instance_name>.*)/)?blobs/(?P<hash>[a-f0-9]{64})/(?P<size>\\d+)")
-	actionCacheRegex = regexp.MustCompile("^(?:(?P<instance_name>.*)/)?blobs/ac/(?P<hash>[a-f0-9]{64})/(?P<size>\\d+)")
+	uploadRegex      = regexp.MustCompile(`^(?:(?:(?P<instance_name>.*)/)?uploads/(?P<uuid>[a-f0-9-]{36})/)?(?P<blob_type>blobs|compressed-blobs/zstd)/(?P<hash>[a-f0-9]{64})/(?P<size>\d+)`)
+	downloadRegex    = regexp.MustCompile(`^(?:(?P<instance_name>.*)/)?(?P<blob_type>blobs|compressed-blobs/zstd)/(?P<hash>[a-f0-9]{64})/(?P<size>\d+)`)
+	actionCacheRegex = regexp.MustCompile(`^(?:(?P<instance_name>.*)/)?(?P<blob_type>blobs|compressed-blobs/zstd)/ac/(?P<hash>[a-f0-9]{64})/(?P<size>\d+)`)
 )
 
-type InstanceNameDigest struct {
+type ResourceName struct {
 	*repb.Digest
 	instanceName string
+	compressor   repb.Compressor_Value
 }
 
-func NewInstanceNameDigest(d *repb.Digest, instanceName string) *InstanceNameDigest {
-	return &InstanceNameDigest{
+func NewResourceName(d *repb.Digest, instanceName string, compressor repb.Compressor_Value) *ResourceName {
+	return &ResourceName{
 		Digest:       d,
 		instanceName: instanceName,
+		compressor:   compressor,
 	}
 }
 
-func (i *InstanceNameDigest) GetInstanceName() string {
-	return i.instanceName
+func NewInstanceNameDigest(d *repb.Digest, instanceName string) *ResourceName {
```

Comment:
we should call this something else -- NewUncompressedResource?

Reply:
Resolved offline to call this `NewResourceName` and then have a separate method for setting the compressor, since it's much less common for compression to be set at all.